### DISCUSSION TITLE
How to redact headers information

### DISCUSSION THREAD

**Discussion by mkarthiatgithub** [aws-powertools/powertools-lambda-python#3284] [General]
Hi, 
Started using powertools and so far so good. Thank you ! 
While printing the lambda events, its as well printing headers and multivalue headers, which got some sensitive data. 
I could not remove them using logger.remove_keys. 
please if there are any ways, help me out.

**Comment by leandrodamascena** [comment]
Hi @mkarthiatgithub! It's good to know that you're enjoying using Powertools!

Just to make sure I understand the question, I believe you are using `@logger.inject_lambda_context(log_event=True)` right? In this case, I understand your point about sensitive data, but currently there is no way to remove the keys. 
We advise in our [documentation](https://docs.powertools.aws.dev/lambda/python/latest/core/logger/#logging-incoming-event) this feature should not be used in production environments as it is for debugging purposes only.

Do you need to print the event in production? Why not using our [Event Source data class](https://docs.powertools.aws.dev/lambda/python/latest/utilities/data_classes/) utility and print only specific fields?

Thanks

**Reply by mkarthiatgithub** [reply]
Thank you , this certainly helped and I could able to use Event Source data class !
